### PR TITLE
dev/drupal#52 Override getUrlPath() for Drupal8

### PR DESCRIPTION
**Pull request by mlutfy**
Overview
----------------------------------------

Aims to deprecate the use of the "q" variable in Drupal8. The "q" variable is deprecated, so accessing it throws notices/warnings.

More details: https://lab.civicrm.org/dev/drupal/issues/52

This is a companion PR for #15265

Full disclosure: I have been running a variant of this patch in production for a long time, but I have not tested this specific patch.

### CODE REVIEW

File: CRM/Utils/System.php [hunk 1]
```diff
@@ -306,8 +306,14 @@ public static function url(
...
+
+    if (method_exists($config->userSystem, 'getUrlPath')) {
+      return $config->userSystem->getUrlPath();
+    }
+
+    if (isset($_GET[$config->userFrameworkURLVar])) {
+      return $_GET[$config->userFrameworkURLVar];
     }
     return NULL;
   }
```
Comment (totten) [inline CRM/Utils/System.php]: I have a small `r-code` suggestion to make this polymorphism more conventional. Will push up onto this PR in a few minutes after `r-running` on D7 and writing commit notes.

Comment (totten) [reply CRM/Utils/System.php]: Pushed up. Tested on D7 to ensure that the refactoring was generally well-formed. Discussed more in commit notes.

Comment (mlutfy) [reply CRM/Utils/System.php]: Looks good. Thank you!

**Comment by seamuslee001** [comment]
is this still needed @mlutfy ?

**Comment by mlutfy** [comment]
@seamuslee001 I think it is. I will try to review again, now that I have finally upgraded to Drupal 8.8.

**Comment by jaapjansma** [comment]
Betty and I are reviewing PRs and we came across this one. We see the latest work has been done september 11 therefor we assume this PR can be closed as it needs work and nothing happens on it.

Feel free to reopen the PR if needed.

**Comment by mlutfy** [comment]
Hi Jaap and Better, thanks for the review. However, #15268 was only a partial fix. This patch is still required.

**Comment by mlutfy** [comment]
I updated my branch to resolve the conflicts.

Apologies for the typo in my last comment.

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by jaapjansma** [comment]
@mlutfy the test is failing because of style warnings are you able to fix them?

**Comment by jaapjansma** [comment]
test this please

**Comment by mattwire** [comment]
@mlutfy Needs rebase

**Comment by eileenmcnaughton** [comment]
@mlutfy a lot has changed since this + it's stale. I'm gonna close for now - please e-open as appropriate

**Pull request closed without merging**